_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,373 @@
+
+```json
+POST _ingest/pipeline/dot-expander-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "user.address.city": "New York",
+        "user.address.state": "NY"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "user": {
+            "address": {
+              "city": "New York",
+              "state": "NY"
+            }
+          }
+        },
+        "_ingest": {
+          "timestamp": "2024-01-17T01:32:56.501346717Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+### Step 3: Ingest a document
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=dot-expander-pipeline
+{
+  "user.address.city": "Denver",
+  "user.address.state": "CO"
+}
+```
+{% include copy-curl.html %}
+
+### Step 4 (Optional): Retrieve the document
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response confirms that the specified fields were expanded into nested fields:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 1,
+  "_seq_no": 3,
+  "_primary_term": 1,
+  "found": true,
+  "_source": {
+    "user": {
+      "address": {
+        "city": "Denver",
+        "state": "CO"
+      }
+    }
+  }
+}
+```
+
+## The `path` parameter
+
+You can use the `path` parameter to specify the path to a dotted field within an object. For example, the following pipeline specifies the `address.city` field that is located within `user` object: 
+
+```json
+PUT /_ingest/pipeline/dot-expander-pipeline
+{
+  "description": "Dot expander processor",
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "address.city",
+        "path": "user"
+      }
+    },
+    {
+      "dot_expander":{
+       "field": "address.state",
+       "path": "user"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Simulate the pipeline as follows: 
+
+```json
+POST _ingest/pipeline/dot-expander-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "user": {
+          "address.city": "New York",
+          "address.state": "NY"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The `dot_expander` processor transforms the document into:
+
+```json
+{
+  "user": {
+    "address": {
+      "city": "New York",
+      "state": "NY"
+    }
+  }
+}
+```
+
+## Field name conflicts
+
+If there already exists a field with the same path as the path where the `dot_expander` processor should expand the value, the processor merges the two values into an array.
```

Comment:
Should "where" be "to which"? Unsure of this one.

Reply:
Revised